packages/strapi/lib/middlewares/responses/index.js

```diff
@@ -22,8 +22,8 @@ module.exports = () => {
         }
 
         // Set X-Powered-By header.
-        if (_.get(strapi.config, 'X-Powered-By.enabled', true)) {
-          ctx.set('X-Powered-By', 'Strapi <strapi.io>');
+        if (_.get(strapi.config.currentEnvironment.response, 'poweredBy.enabled', null)) {
```

Comment:
You don't need to set `null` as default value here.
If there is no value, it will be `undefined` and it's a good value to not enter in the condition.